src/elfcore.cc

```diff
@@ -1190,25 +1376,33 @@ static int CreateElfCore(void *handle, ssize_t (*writer)(void *, const void *, s
               }
             }
           }
-          /* The order of threads in the output matters to gdb:
-           * it assumes that the first one is the one that crashed.
-           * Make it easier for the end-user to find crashing thread
-           * by dumping it first.
-           */
-          for (i = num_threads; i-- > 0;) {
-            if (pids[i] == main_pid) {
-              if (WriteThreadRegs(handle, writer, prstatus, pids[i], regs + i, fpregs + i, fpxregs + i)) {
-                goto done;
+          if (!ext_process) {
+              /* The order of threads in the output matters to gdb:
+               * it assumes that the first one is the one that crashed.
+               * Make it easier for the end-user to find crashing thread
+               * by dumping it first.
+               */
+              for (i = num_threads; i-- > 0;) {
+                if (pids[i] == main_pid) {
+                  if (WriteThreadRegs(handle, writer, prstatus, pids[i], regs + i, fpregs + i, fpxregs + i)) {
+                    goto done;
+                  }
+                  break;
+                }
               }
-              break;
-            }
-          }
-          for (i = num_threads; i-- > 0;) {
-            if (pids[i] != main_pid) {
-              if (WriteThreadRegs(handle, writer, prstatus, pids[i], regs + i, fpregs + i, fpxregs + i)) {
-                goto done;
+              for (i = num_threads; i-- > 0;) {
+                if (pids[i] != main_pid) {
+                  if (WriteThreadRegs(handle, writer, prstatus, pids[i], regs + i, fpregs + i, fpxregs + i)) {
+                    goto done;
+                  }
+                }
+              }
+          } else {
+              for (i = num_threads; i-- > 0;) {
```

Comment:
why are we not doing the main_pid thing here? Surely it still matters?

Reply:
Yes, we should. Done.